libs/wingsdk/src/target-sim/function.inflight.ts

```diff
@@ -40,24 +58,24 @@ export class Function implements IFunctionClient, ISimulatorResourceInstance {
     return this.context.withTrace({
       message: `Invoke (payload=${JSON.stringify(payload)}).`,
       activity: async () => {
-        const vm = new NodeVM({
-          console: "redirect", // we hijack `console.xxx` in `cloud/function.ts`
-          require: {
-            external: true,
-            builtin: ["*"], // allow using all node modules
-            context: "sandbox", // require inside the sandbox (addresses #1871)
-          },
-          sandbox: {
-            $simulator: this.context,
-          },
-          env: {
-            ...process.env,
-            ...this.env,
+        if (!this.bundle) {
+          throw new Error("unable to find bundle (function not initialized?)");
+        }
+
+        const sandboxProcess = {
+          ...process,
+
+          // override process.exit to throw an exception instead of exiting the process
+          exit: (exitCode: number) => {
+            throw new Error(
+              "process.exit() was called with exit code " + exitCode
+            );
           },
-          timeout: this.timeout,
-        });
 
-        // see https://github.com/patriksimek/vm2/blob/master/lib/nodevm.js#L89
+          env: this.env,
+        };
+
+        const sandboxConsole: any = {};
         const levels = [
```

Comment:
Maybe consider removing "dir" and "trace" from this list (or add a TODO to handle them differently) -- it looks like we changing the behavior of the functions so that they no longer emit a list of properties or a stack trace respectively, which might be surprising

Reply:
Done. Thanks!